runtime/router.go

```diff
@@ -251,3 +262,39 @@ func (router *Router) handleMethodNotAllowed(
 		http.StatusMethodNotAllowed,
 	)
 }
+
+func (router *Router) logRequest(r *http.Request) error {
+	// TODO: Allocating a fixed size array causes the zap logger to fail
+	// with ``unknown field type: { 0 0  <nil>}'' errors. Investigate this
+	// further to see if we can avoid reallocating underlying arrays for slices.
+	fields := make([]zapcore.Field, 0)
+	for k, v := range r.Header {
+		if len(v) > 0 {
+			fields = append(fields, zap.String(k, v[0]))
+		}
+	}
+
+	fields = append(fields, zap.String("method", r.Method))
+	fields = append(fields, zap.Int64("content-length", r.ContentLength))
+	fields = append(fields, zap.String("host", r.Host))
+	body, err := ioutil.ReadAll(r.Body)
+	r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+
+	fields = append(fields, zap.String("body", string(body)))
```

Comment:
Removed for now, we should add the ability for gateway owner to define custom marshallers that don't log PII